app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js

```diff
@@ -201,13 +204,13 @@ export async function switchChain(
       if (!ethChainIds.includes(chainId)) {
         await requestPermittedChainsPermissionIncrementalForOrigin({
           chainId,
-          autoApprove: isAddFlow,
+          autoApprove: isAddFlow || isSnapId(origin),
```

Comment:
i have a PR here that replaces the `isAddFlow` param with one named `autogrant`. Perhaps this isSnapId check can happen outside the `switchChain` helper that way we don't need to pass `origin` in?

Reply:
Makes sense. I didn't want to use `isAddFlow` since it's not an add flow, but `autoApprove` sounds good.

Reply:
i managed to not link the PR lol. Here it is

https://github.com/MetaMask/metamask-extension/pull/29837